conan/cli/commands/compare.py

```diff
@@ -0,0 +1,168 @@
+import os
+
+from conan.api.output import ConanOutput
+from conan.cli.formatters.compare.compare import format_compare_html, format_compare_txt, \
+    format_compare_json
+from conan.errors import ConanException
+from conan.api.conan_api import ConanAPI
+from conan.cli.command import conan_command
+from conan.api.model import RecipeReference
+from conan.internal.conan_app import ConanApp
+from conan.internal.errors import conanfile_exception_formatter
+from conan.internal.graph.graph import CONTEXT_HOST
+from conan.internal.graph.profile_node_definer import initialize_conanfile_profile
+from conan.internal.source import retrieve_exports_sources, config_source
+from conan.internal.util.files import rmdir
```

Comment:
The command layer shouldn't use anymore any ``conan.internal`` details, so all of this must be hidden in the API.
We need to find:
- Command name
- Location in the API